lib/async_hooks.js

```diff
@@ -211,6 +211,13 @@ class AsyncResource {
   triggerAsyncId() {
     return this[trigger_async_id_symbol];
   }
+
+  static bind(fn, type = fn?.name) {
+    const asyncResource = new AsyncResource(type || 'bound-anonymous-fn');
+    return asyncResource.runInAsyncScope.bind(asyncResource, (...args) => {
+      fn(...args);
```

Comment:
return value not preserved?

Reply:
actually couldn't this just be `asyncResource.runInAsyncScope.bind(asyncResource, fn)`?

Reply:
Yeah, was premature about opening the PR.. had a few local changes not checked in lol... will get those in soon ;-)